types/mithril/index.d.ts

```diff
@@ -208,7 +208,7 @@ declare namespace Mithril {
 	 * Any Javascript object that has a view method can be used as a Mithril component.
 	 * Components can be consumed via the m() utility.
 	 */
-	interface Component<Attrs, State extends Lifecycle<Attrs, State>> extends Lifecycle<Attrs, State> {
```

Comment:
@isiahmeadows can you provide example usage of type inference? (Even better, if you have time, a PR for the mithril tests would serve as permanent documentation.) mithril's usage sounds like it could be the same as vue's &mdash; our work to support vue resulted in `ThisType<T>`. But it affects contextual typing, so it only works in calls, where both type inference and contextual typing are in play.